vassal-app/src/main/java/VASSAL/script/ExpressionInterpreter.java

```diff
@@ -636,17 +636,33 @@ public Object sum(Object src, Object propertyName, Object propertyMatch) {
 
   public Object sum(Object src, Object propertyName, Object propertyMatch, Object mapName) {
     int result = 0;
+    List<Map> mapList = new ArrayList<>();
 
-    if (! (src instanceof GamePiece)) return 0;
+    if (! (src instanceof PropertySource)) return 0;
     if (! (propertyName instanceof String)) return 0;
     if (! (propertyMatch == null || propertyMatch instanceof String)) return 0;
     if (! (mapName == null || mapName instanceof String)) return 0;
 
-    final GamePiece sourcePiece = (GamePiece) src;
-    final String matchString = replaceDollarVariables((String) propertyMatch, sourcePiece);
-    final PieceFilter filter = matchString == null ? null : new PropertyExpression(unescape(matchString)).getFilter(sourcePiece);
+    final String matchString = replaceDollarVariables((String) propertyMatch, (PropertySource) src);
+    final PieceFilter filter = matchString == null ? null : new PropertyExpression(unescape(matchString)).getFilter((PropertySource) src);
 
-    for (final Map map : getMapList(mapName, sourcePiece)) {
+    if (src instanceof GamePiece) {
+      mapList = getMapList(mapName, (GamePiece) src);
+    }
+    else if (mapName != null && !((String) mapName).isEmpty()) {
```

Comment:
Is the piece always the source in spite of a passed "mapName"? Or is a mapName sometimes passed on a piece (for some other reason) and doesn't take over as the map but does do other stuff?

Reply:
This is the original behavior (lines 649-651) that I have left unchanged as the first check. The GamePiece, if supplied, is just used as the property source to build the piece Filter and provide the default map to visit. The mapName, usually null, is used to select an alternate Map to visit.